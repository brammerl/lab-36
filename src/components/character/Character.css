.Character {
  display: grid;
  grid-template-columns: 1fr 5fr;
  transition: .2s;

  img {
    width: 10rem;
  }

  
}

.Character-light{
  composes: Character;

  background-color: white;
  color: black;
}

.Character-dark {
  composes: Character;

  background-color: darkgrey;
  color: white;
}