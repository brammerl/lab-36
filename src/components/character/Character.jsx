import React from 'react';
import { useCharacter, useTheme } from '../../hooks/appContext';
import styles from './Character.css';

const Character = () => {
  const character = useCharacter();
  const theme = useTheme();

  if(!character) return <h1>Loading!</h1>;

  const characterList = character.map(character => {
  return (<figure className={styles[`Character-${theme}`]}>
      <img src={character.photoUrl} alt={character.name}/>
      <figcaption>{character.name}</figcaption>
    </figure>)
  });

  return (
  <>
    {characterList}
  </>
  )
}

export default Character;