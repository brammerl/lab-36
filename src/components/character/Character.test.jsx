import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character Component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  