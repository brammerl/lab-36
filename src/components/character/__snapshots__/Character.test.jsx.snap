// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component renders Character 1`] = `
<h1>
  Loading!
</h1>
`;
